File: src/main/scala/assignment1/Group.scala
package assignment1

class Group[T] {
  private var contents = List[T]()

  def add(x: T): Unit = {
    if (!has(x)) {
      contents = x :: contents 
    }
  }

  def delete(x: T): Unit = {
    contents = contents.filterNot(_ == x) 
  }

  def has(x: T): Boolean = {
    contents.contains(x) 
  }
}

object Group {
  def from[T](xs: Iterable[T]): Group[T] = {
    val result = new Group[T]
    for (x <- xs) {
      result.add(x) 
    }
    result
  }
}

@main def test() = {
  val group = Group.from(List(10, 20))
  println(group.has(10))
  // → true
  println(group.has(30))
  // → false
  group.add(10)
  group.delete(10)
  println(group.has(10))
  // → false
}

File: src/main/js/assignment1/Group.js
class Group {
    mySet;

    constructor() {
        this.mySet = [];
    }

    has(x) {
        return this.mySet.includes(x);
    }

    add(x) {
        if (!this.has(x)) {
            this.mySet.push(x);
        }
    }

    delete(x) {
        this.mySet = this.mySet.filter(item => item !== x);;
    }

    static from(it) {
        let result = new Group();
        for (let x of it) {
            result.add(x);
        }
        return result;
    }
}

function main() {
    let group = Group.from([10, 20]);
    console.log(group.has(10));
    // → true
    console.log(group.has(30));
    // → false
    group.add(10);
    group.delete(10);
    console.log(group.has(10));
    // → false
}

if (require.main === module) {
    main();
}

module.exports = Group;
